source/base/parameter_acceptor.cc

```diff
@@ -182,9 +191,37 @@ ParameterAcceptor::get_section_path() const
             break;
           }
     }
+  // Finally, insert the remaining subsections
+  sections.insert(sections.end(), subsections.begin(), subsections.end());
   return sections;
 }
 
+
+
+void
+ParameterAcceptor::enter_subsection(const std::string &subsection)
+{
+  AssertThrow(subsection.find(sep) == std::string::npos,
+              ExcMessage(
+                "A subsection name cannot contain the special character '/'"));
+  // First the easy case.
+  if (subsection == "")
+    return;
```

Comment:
I think you should throw an exception in this case. If you just return here, then user code will no longer have matching enter/leave subsection calls.

Reply:
done.